pkg/stanza/fileconsumer/internal/reader/reader.go

```diff
@@ -30,18 +30,20 @@ type Metadata struct {
 // Reader manages a single file
 type Reader struct {
 	*Metadata
-	logger          *zap.SugaredLogger
-	fileName        string
-	file            *os.File
-	fingerprintSize int
-	maxLogSize      int
-	lineSplitFunc   bufio.SplitFunc
-	splitFunc       bufio.SplitFunc
-	decoder         *decode.Decoder
-	headerReader    *header.Reader
-	processFunc     emit.Callback
-	emitFunc        emit.Callback
-	deleteAtEOF     bool
+	logger                 *zap.SugaredLogger
+	fileName               string
+	file                   *os.File
+	fingerprintSize        int
+	initialBufferSize      int
```

Comment:
Am I missing something or is the `initialBufferSize` field only used by the fingerprint tests?

It seems the value of this outside of tests is always equal to `scanner.DefaultBufferSize`, is this true?

Reply:
I think you are correct and typically wouldn't want to have such a field. However, I think we may want to expose this field to user configuration eventually as well. In a scenario where there are many files being watched, each of which tends to contain only short lines, this could save substantial memory. In the meantime, my thinking is that it's necessary to validate behaviors which we do not have control over (buffer size after initialization).

Reply:
> it's necessary to validate behaviors which we do not have control over

I :100:  agree in principle. In this case, it is lost on me why the unit tests need to be run with various values for the initial buffer size, when in real life the initial buffer size is always equal to 16k.

Anyway, I don't think this should be blocking this PR.

Reply:
> it is lost on me why the unit tests need to be run with various values for the initial buffer size, when in real life the initial buffer size is always equal to 16k.

Maybe there is a better way but I'm just trying to make sure the code is robust because we only have control over the _initial_ buffer size. In real life, the buffer is resized automatically and I don't want to make assumptions about exactly which values it may take. Most of the test cases attempt to create boundary conditions by reading controlled inputs through various small sizes. I think odds of hitting these boundary conditions are much lower if only working against 16k+ buffers.